File: svc-data/src/main/java/com/beancounter/marketdata/assets/figi/FigiSearch.java
package com.beancounter.marketdata.assets.figi;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class FigiSearch {
  @Builder.Default
  private String idType = "BASE_TICKER";
  private String idValue;
  private String exchCode;
  private String securityType2;
}

File: svc-data/src/main/java/com/beancounter/marketdata/assets/figi/FigiSearch.kt
package com.beancounter.marketdata.assets.figi

/**
 * Search params into OpenFigi.
 */
data class FigiSearch(
    var idValue: String,
    var exchCode: String,
    var securityType2: String = "Common Stock",
    val includeUnlistedEquities: Boolean = true
) {
    var idType = "BASE_TICKER"
    var isIncludeUnlistedEquities = true

    init {
        isIncludeUnlistedEquities = includeUnlistedEquities
    }
}
